vercel/resource_edge_config_item.go

```diff
@@ -0,0 +1,260 @@
+		Key:   types.StringValue(out.Key),
+		Value: types.StringValue(out.Value),
+	}
+}
+
+// Create will create an edgeConfigToken within Vercel.
+// This is called automatically by the provider when a new resource should be created.
+func (r *edgeConfigItemResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan EdgeConfigItem
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := r.client.CreateEdgeConfigItem(ctx, client.CreateEdgeConfigItemRequest{
+		TeamID:       plan.TeamID.ValueString(),
+		EdgeConfigID: plan.EdgeConfigID.ValueString(),
+		Token:        plan.Token.ValueString(),
+		Key:          plan.Key.ValueString(),
+		Value:        plan.Value.ValueString(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating Edge Config Item",
+			"Could not create Edge Config Item, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	result := responseToEdgeConfigItem(out)
+	tflog.Info(ctx, "created Edge Config Item", map[string]interface{}{
+		"edge_config_id": plan.EdgeConfigID.ValueString(),
+		"key":            result.Key.ValueString(),
+		"value":          result.Value.ValueString(),
+	})
+
+	diags = resp.State.Set(ctx, result)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read will read edgeConfigToken information by requesting it from the Vercel API, and will update terraform
+// with this information.
+func (r *edgeConfigItemResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state EdgeConfigItem
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := r.client.GetEdgeConfigItem(ctx, client.EdgeConfigItemRequest{
+		Token:        state.Token.ValueString(),
+		TeamID:       state.TeamID.ValueString(),
+		EdgeConfigID: state.EdgeConfigID.ValueString(),
+		Key:          state.Key.ValueString(),
+		Value:        state.Value.ValueString(),
+	})
+	if client.NotFound(err) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading Edge Config Item",
+			fmt.Sprintf("Could not get Edge Config Item %s %s, unexpected error: %s",
+				state.TeamID.ValueString(),
+				state.Key.ValueString(),
+				err,
+			),
+		)
+		return
+	}
+
+	result := responseToEdgeConfigItem(out)
+	tflog.Info(ctx, "read edge config token", map[string]interface{}{
+		"team_id":        state.TeamID.ValueString(),
+		"edge_config_id": state.EdgeConfigID.ValueString(),
+		"key":            state.Key.ValueString(),
+	})
+
+	diags = resp.State.Set(ctx, result)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update is the same as Create
+func (r *edgeConfigItemResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
We should remove the contents of the Update method, making it `panic` instead. And rely on the schema having `RequiresReplace` for every field.